internal/controller/apiextensions/composite/composition_pt.go

```diff
@@ -143,7 +143,7 @@ func NewPTComposer(kube client.Client, o ...PTComposerOption) *PTComposer {
 	kube = unstructured.NewClient(kube)
 
 	c := &PTComposer{
-		client: resource.ClientApplicator{Client: kube, Applicator: resource.NewAPIPatchingApplicator(kube)},
+		client: resource.ClientApplicator{Client: kube, Applicator: resource.NewAPIUpdatingApplicator(kube)},
```

Comment:
I don't think we can simply switch to the updating applicator from the patching applicator without breaking some existing cases. I know, at least in the Usage implementation, I [rely on this fact](https://github.com/crossplane/crossplane/pull/4444/files#diff-539ca26eb860a34f63a58b7896e1287da46840b0a6f144b20c14cdf79ac71c38R330-R332) not to get labels put by the Usage controller deleted by composition controller. 

There could be other cases that rely on this fact that need to be handled properly, including the ones we know, e.g. what would happen to external-name annotation put on composed resources by providers, and we don't know, e.g. some external/custom controller adding some custom metadata on composed resources. In that sense, this could be considered as a behavioral breaking change.